features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Create/Default/APICreateDefault.jsx

```diff
@@ -358,13 +428,18 @@ function APICreateDefault(props) {
                                     id='itest-id-apicreatedefault-createnpublish'
                                     variant='contained'
                                     color='primary'
-                                    disabled={!isPublishable || isAPICreateDisabled || !apiInputs.isFormValid}
+                                    disabled={isDeploying || isRevisioning || !isPublishable
+                                        || isAPICreateDisabled || !apiInputs.isFormValid}
                                     onClick={createAndPublish}
                                 >
-                                    {!isPublishing && 'Create & Publish'}
-                                    {isPublishing && <CircularProgress size={24} />}
+                                    {(!isPublishing && !isRevisioning && !isDeploying) && 'Create & Publish'}
+                                    {(isPublishing || isRevisioning || isDeploying) && <CircularProgress size={24} />}
                                     {isCreating && isPublishing && 'Creating API . . .'}
-                                    {!isCreating && isPublishing && 'Publishing API . . .'}
+                                    {!isCreating && isRevisioning && !isDeploying && 'Creating Revision . . .'}
+                                    {!isCreating && isPublishing
+                                        && !isRevisioning && !isDeploying && 'Publishing API . . .'}
```

Comment:
Can't we simply use the respective action state to show the relevant text 

I:e

```js
isCreating && {'Creating API . . .'}
```

likewise ?